api/app/tests/apps/api/test_api_v1_endpoints.py

```diff
@@ -1378,6 +1378,73 @@ def test_split_more_than_max_data(self):
 
         self.assertEqual(Signal.objects.count(), 2)
 
+    def test_update_location_renders_correctly_in_history(self):
+        """Test that location updates have correct description field in history.
+
+        Valid addresses in SIA have:
+        - coordinates and address
+        - only cooordinates
+
+        Furthermore locations without stadsdeel property should render as well.
+        """
+        self.client.force_authenticate(user=self.sia_read_write_user)
+        detail_endpoint = self.detail_endpoint.format(pk=self.signal_no_image.id)
+
+        # Prepare the data for the 3 types of Location updates
+        update_location_json = os.path.join(THIS_DIR, 'update_location.json')
+        with open(update_location_json, 'r') as f:
+            data_with_address = json.load(f)
+        data_no_address = copy.deepcopy(data_with_address)
+        data_no_address['location']['address'] = {}
+        data_no_address_no_stadsdeel = copy.deepcopy(data_no_address)
+        data_no_address_no_stadsdeel['location']['stadsdeel'] = None
+
+        history_endpoint = self.history_endpoint.format(pk=self.signal_no_image.id)
+        # Test full location (address and coordinates) case:
+        response = self.client.patch(detail_endpoint, data=data_with_address, format='json')
+        self.assertEqual(response.status_code, 200)
+
+        response = self.client.get(history_endpoint + '?what=UPDATE_LOCATION')
+        response_data = response.json()
+        self.assertEqual(len(response_data), 2)
+        self.assertEqual(
+            'Stadsdeel: Centrum\nDe Ruijterkade 36A\nAmsterdam',
+            response_data[0]['description']
+        )
+
+        # Test no address case:
+        response = self.client.patch(detail_endpoint, data=data_no_address, format='json')
+        self.assertEqual(response.status_code, 200)
+
+        response = self.client.get(history_endpoint + '?what=UPDATE_LOCATION')
+        response_data = response.json()
+        self.assertEqual(len(response_data), 3)
+        # import pprint
+        # pprint.pprint(response_data)
+        # pprint.pprint(data_no_address)
+
+        # import pdb; pdb.set_trace()
```

Comment:
Remove the above commented debug statements

Reply:
removed, pushed